source/PreferencesPanel.cpp

```diff
@@ -878,6 +939,76 @@ void PreferencesPanel::DrawPlugins()
 
 
 
+void PreferencesPanel::DrawPluginInstalls()
+{
+	const Color &back = *GameData::Colors().Get("faint");
+	const Color &dim = *GameData::Colors().Get("dim");
+	const Color &medium = *GameData::Colors().Get("medium");
+	const Color &bright = *GameData::Colors().Get("bright");
+	const Color &outdated = *GameData::Colors().Get("plugin outdated");
+
+	const int MAX_TEXT_WIDTH = 230;
+	Table table;
+	table.AddColumn(-115, {MAX_TEXT_WIDTH, Truncate::MIDDLE});
+	table.SetUnderline(-120, 100);
+
+	int firstY = -238;
+	table.DrawAt(Point(-110, firstY));
+	table.DrawUnderline(medium);
+	table.DrawGap(25);
+
+	const Font &font = FontSet::Get(14);
+
+	const size_t currentPageIndex = MAX_PLUGIN_INSTALLS_PER_PAGE * currentPluginInstallPage;
+	const int maxIndex = min(currentPageIndex + MAX_PLUGIN_INSTALLS_PER_PAGE, pluginInstallList.size());
+	for(int x = currentPageIndex; x < maxIndex; x++)
+	{
+		const auto &plugin = pluginInstallList.at(x);
+		const Plugin *installedVersion = Plugins::Get().Find(plugin["name"]);
+		bool isInstalled = installedVersion && !installedVersion->removed;
+		Plugins::InstallData installData(
+			plugin["name"],
+			plugin["url"],
+			plugin["version"],
+			plugin["description"],
+			isInstalled,
+			isInstalled && installedVersion->version != plugin["version"]
+		);
```

Comment:
We're recreating the installation data on every frame and going through the json every time too. Do them after the file has been downloaded and store them in a vector.

Reply:
ok